src/trapfunc.cpp

```diff
@@ -73,7 +73,7 @@ void trapfunc::none( Creature *, const tripoint & )
 void trapfunc::bubble( Creature *c, const tripoint &p )
 {
     // tiny animals don't trigger bubble wrap
-    if( ( c != nullptr && c->get_size() == MS_TINY ) || c->is_hallucination() ) {
+    if( c != nullptr && ( c->get_size() == MS_TINY || c->is_hallucination() ) ) {
```

Comment:
All of the instances in this file would be greatly improved by switching to
```suggestion
    if( c == nullptr ||  c->get_size() == MS_TINY || c->is_hallucination() ) {
```

Reply:
I'm totally wrong, nevermind.

Reply:
It's not universally the right thing to do, but sometimes it is.

Reply:
After taking another look, we can follow up with that later, it's not a total no-brainer to fix.

Reply:
Thank you for your review Kevin! Yes, that's the trick. Even if `Creature` is a null pointer you still need to enter the functions... I was about to make a comment on each functions because if you stare at the trap functions it's not very obvious. I decided not to because it might be a bit annoying for some developers. 
And yes I do agree it probably needs a bit of refactoring in the future.